build_tools/azure/install.sh

```diff
@@ -86,6 +86,13 @@ elif [[ "$DISTRIB" == "ubuntu-32" ]]; then
     python3 -m virtualenv --system-site-packages --python=python3 $VIRTUALENV
     source $VIRTUALENV/bin/activate
     python -m pip install pytest==$PYTEST_VERSION pytest-cov cython joblib==$JOBLIB_VERSION
+elif [[ "$DISTRIB" == "latest"]]
+    # since conda main channel usually lacks behind on the latest releases,
+    # we use pypi to test against the latest releases of the dependencies.
+    make_conda("python=$PYTHON_VERSION")
+    python -m pip install numpy scipy joblib cython
+    python -m pip install pytest==$PYTEST_VERSION pytest-cov pytest-xdist
+    python -m pip install mkl pandas matplotlib pyamg pillow
```

Comment:
Why MKL? Would it even have an effect? scipy and numpy would use the vendored OpenBLAS in this case anyway I think.

Reply:
I just basically created the same environment as we had before. I'm not changing it, before this PR, MKL was included in the "latest" environment.

Reply:
Or is it that if we get them from conda, they would need mkl, but the pypi wheels don't need it?

Reply:
> Or is it that if we get them from conda, they would need mkl, but the pypi wheels don't need it?

Yes, because with conda you can change the BLAS backend to MKL, OpenBLAS or BLIS (at least on conda-forge). However with when pip installing latest pre-build wheels we don't have a choice -- it's going to be OpenBLAS I think.